js/portfolio/articles.js

```diff
@@ -1,87 +1,76 @@
+    inputData.sort(function(a, b) {
+      return (new Date(b.pubDate)) - (new Date(a.pubDate));
+    });
 
-        if (newETag === JSON.parse(localStorage.eTagValue)) {
-          console.log('no change');
-          Article.loadAll(JSON.parse(localStorage.portfolioData));
-          portfolioView.initIndexPage();
+    Article.all = inputData.map(function(item) {
+      return new Article(item);
+    });
+  };
 
-        } else {
-          $.ajax({
-            url: 'data/portfolioData.json',
-            dataType: 'JSON',
-            success: function(data, message, xhr) {
-              Article.loadAll(data);
-              localStorage.eTagValue = JSON.stringify(newETag);
-              localStorage.portfolioData = JSON.stringify(data);
-              console.log('loading new data');
-              portfolioView.initIndexPage();
-            }
-          });
+  Article.getAll = function(next) {
+    $.getJSON('/data/portfolioData.json', function(responseData) {
+      Article.loadAll(responseData);
+      localStorage.portfolioData = JSON.stringify(responseData);
+      next();
+    });
+  };
 
-          // different syntax for experimenting
-          // $.getJSON('data/portfolioData.json', function(jsondata) {
-          //   Article.loadAll(jsondata);
-          //   localStorage.eTagValue = JSON.stringify(newETag);
-          //   localStorage.portfolioData = JSON.stringify(jsondata);
-          //   console.log('loading new data');
-          //   portfolioView.initIndexPage();
-          // });
+  Article.fetchAll = function(next) {
+    if (localStorage.portfolioData) {
+      $.ajax({
+        type: 'HEAD',
+        url: '/data/portfolioData.json',
+        success: function(data, message, xhr) {
+          var eTag = xhr.getResponseHeader('eTag');
+          if (!localStorage.eTag || eTag !== localStorage.eTag) {
+            localStorage.eTag = eTag;
+            Article.getAll(next);
+          } else {
+            Article.loadAll(JSON.parse(localStorage.portfolioData));
+            next();
+          }
         }
-      }
-    });
-  } else {
-    $.ajax({
-      url: 'data/portfolioData.json',
-      dataType: 'JSON',
-      success: function(data, message, xhr) {
-        var eTag = xhr.getResponseHeader('eTag');
-        localStorage.eTagValue = JSON.stringify(eTag);
-        console.log('saved new eTag');
+      });
+    } else {
+      Article.getAll(next);
+    }
+  };
 
-        Article.loadAll(data);
-        console.log('loaded data');
-        localStorage.portfolioData = JSON.stringify(data);
-        portfolioView.initIndexPage();
-        console.log('Loaded and displayed new Data');
+  Article.checkLatestProject = function() {
+    return Article.all.map(function(article) {
+      return {
+        articleName: article.ttl,
+        articleDate: article.pubDate,
+      };
+    })
+    .reduce(function(a, b) {
+      if (a) {
+        if (Date.parse(b.articleDate) > Date.parse(a.articleDate)) {
```

Comment:
If this is what you're doing, you should really use sort or something instead. You should consider refactoring it now you've proven you can reduce().